lib/src/components/NxTransferList/NxTransferList.tsx

```diff
@@ -38,39 +41,63 @@ export default function NxTransferList<T extends string | number>(props: Props<T
     ...attrs
   } = props;
 
+  if (allowReordering && !Array.isArray(selectedItems)) {
+    throw new TypeError('selectedItems must be an array if allowReordering is true');
+  }
+
+  const selectedItemsArray = Array.isArray(selectedItems) ? selectedItems : Array.from(selectedItems);
+
   const availableItemsCountFormatter = availableItemsCountFormatterProp || defaultAvailableItemsCountFormatter,
       selectedItemsCountFormatter = selectedItemsCountFormatterProp || defaultSelectedItemsCountFormatter;
 
-  const groupedItems = useMemo(() => groupBy(item => selectedItems.has(item.id) ? 'selected' : 'available', allItems),
-          [allItems, selectedItems]),
+  const groupedItems = useMemo(() =>
+        !allowReordering
+          ? groupBy(item => selectedItemsArray.includes(item.id) ? 'selected' : 'available', allItems)
+          : {
+            available: allItems.filter(item => !selectedItemsArray.includes(item.id)),
+            selected: selectedItemsArray
+                .map(item => allItems.find(_item => _item.id === item))
```

Comment:
Repeatedly iterating `allItems` is probably slow.  I'd first build a lookup map from id to item and then use that here

Reply:
> Repeatedly iterating `allItems` is probably slow. I'd first build a lookup map from id to item and then use that here

Thanks, this is updated here: d098925532af2c05b55be1e7b1bd141f26e0352a